src/helpers/cells/cell_evaluation.ts

```diff
@@ -18,148 +19,136 @@ import { detectLink } from "../links";
     return toNumber(content, locale);
-  } else if (isBoolean(content)) {
+  }
+  if (isBoolean(content)) {
     return content.toUpperCase() === "TRUE" ? true : false;
   }
   return content;
 }
 
 export function createEvaluatedCell(
-  value: CellValue,
-  localeFormat: LocaleFormat,
-  message?: string
+  fPayload: FPayload,
+  locale: Locale = DEFAULT_LOCALE
 ): EvaluatedCell {
-  const link = detectLink(value);
-  if (link) {
-    return {
-      ..._createEvaluatedCell(
-        parseLiteral(link.label, localeFormat.locale),
-        {
-          format:
-            localeFormat.format ||
-            detectDateFormat(link.label, localeFormat.locale) ||
-            detectNumberFormat(link.label),
-          locale: localeFormat.locale,
-        },
-        message
-      ),
-      link,
-    };
+  const link = detectLink(fPayload.value);
+  if (!link) {
+    return _createEvaluatedCell(fPayload, locale);
   }
-  return _createEvaluatedCell(value, localeFormat, message);
+  const linkPayload = {
+    value: parseLiteral(link.label, locale),
+    format:
+      fPayload.format || detectDateFormat(link.label, locale) || detectNumberFormat(link.label),
+  };
+  return {
+    ..._createEvaluatedCell(linkPayload, locale),
+    link,
+  };
 }
 
-function _createEvaluatedCell(
-  value: CellValue,
-  localeFormat: LocaleFormat,
-  message?: string
-): EvaluatedCell {
-  if (isEvaluationError(value)) {
-    return errorCell(value as string, message);
-  }
-  if (localeFormat.format === PLAIN_TEXT_FORMAT) {
-    return textCell(toString(value), localeFormat);
+function _createEvaluatedCell({ value, format, message }: FPayload, locale: Locale): EvaluatedCell {
+  if (value === null) {
+    return emptyCell(format);
   }
-  if (value === "") {
-    return emptyCell(localeFormat);
+  if (isEvaluationError(value)) {
+    return errorCell(value, message);
   }
   if (typeof value === "number") {
-    if (isDateTimeFormat(localeFormat.format || "")) {
-      return dateTimeCell(value, localeFormat);
+    if (isDateTimeFormat(format || "")) {
+      return dateTimeCell(value, { format, locale });
     }
-    return numberCell(value, localeFormat);
-  }
-  if (value === null) {
-    return numberCell(0, localeFormat);
+    return numberCell(value, { format, locale });
   }
   if (typeof value === "boolean") {
-    return booleanCell(value, localeFormat);
+    return booleanCell(value, format);
   }
-  return textCell((value || "").toString(), localeFormat);
+  return textCell(value, { format, locale });
 }
 
-function textCell(value: string, localeFormat: LocaleFormat): EvaluatedCell {
+function textCell(value: string, { format, locale }: LocaleFormat): EvaluatedCell {
   return {
     type: CellValueType.text,
     value,
-    format: localeFormat.format,
+    format,
     isAutoSummable: true,
     defaultAlign: "left",
-    formattedValue: formatValue(value, localeFormat),
+    formattedValue: formatValue(value, { format, locale }),
```

Comment:
we unpack the localeFormat argument, and recreate a new LocaleFormat object  just after. It's a bit of a waste of memory.

Same in other helpers of this file.